### ISSUE TITLE
'year month' format doesn't parse correctly

### ISSUE THREAD

**Issue by dmarx** [scrapinghub/dateparser#527]
Probably related to issue #424. Given a date 'YYYY-MM' format, the parser seems to wrongly assume it's seeing 'YYYY-DD' format and guesses the month as the current month. Examples (executed on June 3rd, 2019):

```
>>> dateparser.parse('1996-04')
datetime.datetime(1996, 6, 4, 0, 0)
>>> dateparser.parse('2010-05')
datetime.datetime(2010, 6, 5, 0, 0)
```

**Comment by kishan3** [comment]
Hi! 
Not able to reproduce the issue.
```
>>> dateparser.parse('1996-04')
datetime.datetime(1996, 4, 9, 0, 0)
>>> dateparser.parse('2010-05')
datetime.datetime(2010, 5, 9, 0, 0)

```
dateparser version: 0.7.1

**Comment by asadurski** [comment]
I must say I can't reproduce this myself. I tried:
- using `RELATIVE_BASE` setting set to `datetime(2019, 6, 3)`,
- rolling back to commits `ac212f45096c2d3905207853b12555a2188f21c5` and `11a761c99d3ee522a3c63756b70c106a579e8b5c` which would be likely ones,
- playing Marty McFly, I set my system date and regional settings to match June 3rd, 2019 in US...
Proof that Github played along: 
![image](https://user-images.githubusercontent.com/7642245/64850884-92add080-d616-11e9-8489-89037ea94fad.png)

None of those tricks worked, no repro. If we can't get to the root of this issue, I guess we should close it and wait for it to resurface.
Perhaps we are missing some system details? Care to update the issue with any more details, @dmarx?
